daemon/cluster/convert/container.go

```diff
@@ -322,16 +327,19 @@ func containerToGRPC(c *types.ContainerSpec) (*swarmapi.ContainerSpec, error) {
 		}
 
 		if m.BindOptions != nil {
-			if mountPropagation, ok := swarmapi.Mount_BindOptions_MountPropagation_value[strings.ToUpper(string(m.BindOptions.Propagation))]; ok {
-				mount.BindOptions = &swarmapi.Mount_BindOptions{Propagation: swarmapi.Mount_BindOptions_MountPropagation(mountPropagation)}
-			} else if string(m.BindOptions.Propagation) != "" {
-				return nil, fmt.Errorf("invalid MountPropagation: %q", m.BindOptions.Propagation)
+			mount.BindOptions = &swarmapi.Mount_BindOptions{
+				// Negative value corresponds to the empty ("") propagation.
+				// Note that zero-value stands for rprivate, which differs from the empty ("") propagation.
+				// See daemon/{oci_linux.go,volumes_linux.go} .
+				Propagation:  -1,
```

Comment:
I tried to use `nil *int32` instead, but `gogoproto.nullable = true` doesn't seems to work here

Reply:
Is that a bug?

Reply:
While technically true that `""` is not the same as `"rprivate"`, I think tht was the intention of the API. Default to rprivate mounts.
I guess this limits what the implementation can do?

Reply:
Defaulting to rprivate is likely to break compatibility

Reply:
> Is that a bug?

Not sure, probably this is expected behavior for `enum`

Reply:
Default should already be rprivate.

Reply:
Hey @AkihiroSuda I'm still not understanding.
Currently the default is to use rprivate, so I don't understand the breaking part.

Reply:
Hmm, why does `moby/daemon/{oci,volume}_linux.go` compare it against an empty string?

Reply:
It is checking if we need to set the propagation.
So in the case of volumes_linux.go we are checking if we need to error out because the user set some unsupported propagation (only slave/shared mounts for paths within the daemon root are supported).
`""` is the default value, which historically means `private`, but in the case of the daemon root, if the user did not set anything explicitly we go ahead and change the propagation to slave to ensure we don't run into other problems.

Reply:
Hope that makes sense.

Reply:
thanks for explanation, I marked this PR as WIP

Reply:
addressed